pkg/store/bucket_test.go

```diff
@@ -1897,3 +1897,190 @@ func createBlockWithOneSeriesWithStep(t testutil.TB, dir string, lbls labels.Lab
+	testutil.Ok(t, head2.Close())
+
+	for _, blockID := range []ulid.ULID{block1, block2} {
+		_, err := metadata.InjectThanos(logger, filepath.Join(bktDir, blockID.String()), thanosMeta, nil)
+		testutil.Ok(t, err)
+	}
+
+	// Instance a real bucket store we'll use to query back the series.
+	fetcher, err := block.NewMetaFetcher(logger, 10, instrBkt, tmpDir, nil, nil, nil)
+	testutil.Ok(tb, err)
+
+	indexCache, err := storecache.NewInMemoryIndexCacheWithConfig(logger, nil, storecache.InMemoryIndexCacheConfig{})
+	testutil.Ok(tb, err)
+
+	store, err := NewBucketStore(
+		logger,
+		nil,
+		instrBkt,
+		fetcher,
+		tmpDir,
+		indexCache,
+		nil,
+		1000000,
+		NewChunksLimiterFactory(10000/MaxSamplesPerChunk),
+		false,
+		10,
+		nil,
+		false,
+		true,
+		DefaultPostingOffsetInMemorySampling,
+		true,
+	)
+	testutil.Ok(tb, err)
+	testutil.Ok(tb, store.SyncBlocks(context.Background()))
+
+	type labelNamesValuesCase struct {
+		name string
+
+		labelNamesReq      *storepb.LabelNamesRequest
+		expectedNames      []string
+		expectedNamesHints hintspb.LabelNamesResponseHints
+
+		labelValuesReq      *storepb.LabelValuesRequest
+		expectedValues      []string
+		expectedValuesHints hintspb.LabelValuesResponseHints
+	}
+
+	testCases := []labelNamesValuesCase{
+		{
+			name: "querying a range containing 1 block should return 1 block in the labels hints",
+
+			labelNamesReq: &storepb.LabelNamesRequest{
+				Start: 0,
+				End:   1,
+			},
+			expectedNames: labelNamesFromSeriesSet(seriesSet1),
+			expectedNamesHints: hintspb.LabelNamesResponseHints{
+				QueriedBlocks: []hintspb.Block{
+					{Id: block1.String()},
+				},
+			},
+
+			labelValuesReq: &storepb.LabelValuesRequest{
+				Label: "__name__",
+				Start: 0,
+				End:   1,
+			},
+			expectedValues: []string{},
+			expectedValuesHints: hintspb.LabelValuesResponseHints{
+				QueriedBlocks: []hintspb.Block{
+					{Id: block1.String()},
+				},
+			},
+		},
+		{
+			name: "querying a range containing multiple blocks should return multiple blocks in the response hints",
+
+			labelNamesReq: &storepb.LabelNamesRequest{
+				Start: 0,
+				End:   3,
+			},
+			expectedNames: labelNamesFromSeriesSet(
+				append(append([]*storepb.Series{}, seriesSet1...), seriesSet2...),
+			),
+			expectedNamesHints: hintspb.LabelNamesResponseHints{
+				QueriedBlocks: []hintspb.Block{
+					{Id: block1.String()},
+					{Id: block2.String()},
+				},
+			},
+
+			labelValuesReq: &storepb.LabelValuesRequest{
+				Label: "__name__",
+				Start: 0,
+				End:   3,
+			},
+			expectedValues: []string{},
+			expectedValuesHints: hintspb.LabelValuesResponseHints{
+				QueriedBlocks: []hintspb.Block{
+					{Id: block1.String()},
+					{Id: block2.String()},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			namesResp, err := store.LabelNames(context.Background(), tc.labelNamesReq)
+			testutil.Ok(t, err)
+			testutil.Equals(t, tc.expectedNames, namesResp.Names)
```

Comment:
This fails with:
```
        bucket_test.go:2055: bucket_test.go:2055:
            
                exp: []string{"ext1", "foo", "i"}
            
                got: []string{"foo", "i"}
            
            Diff:
            --- Expected
            +++ Actual
            @@ -1,3 +1,2 @@
            -([]string) (len=3) {
            - (string) (len=4) "ext1",
            +([]string) (len=2) {
              (string) (len=3) "foo",
```


As in, the `extLabels` we passed are not being returned. From what I know, they should be, but I am not even sure what they are. /cc @pracucci @pstibrany